### PR TITLE
Fixed a possible `nil` pointer dereference if `gracePeriod` is `nil`

### PR DESCRIPTION
**Pull request by disc**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Bug fix

#### Does this PR introduce a user-facing change?

NONE
```release-note
Fix a nil pointer dereference in api server when gracePeriod was set to nil.
```

### CODE REVIEW

**Comment by mouuii** [comment]
@disc  you can find some one to review your code

File: pkg/apis/core/validation/validation_test.go [hunk 1]
```diff
@@ -24148,37 +24148,56 @@ func TestValidateSleepAction(t *testing.T) {
...
+		gracePeriod *int64
 		expectErr   field.ErrorList
 	}{
 		{
 			name: "valid setting",
 			action: &core.SleepAction{
 				Seconds: 5,
 			},
-			gracePeriod: 30,
+			gracePeriod: func() *int64 {
```
Comment (sttts) [inline pkg/apis/core/validation/validation_test.go:24159]: there is `ptr.To[T]` for this

**Comment by sttts** [comment]
Sgtm, but needs somebody from @kubernetes/api-reviewers.

File: pkg/apis/core/validation/validation_test.go (hunk 1 above)
Comment (disc) [reply pkg/apis/core/validation/validation_test.go]: Fixed

**Comment by SergeyKanzhelev** [comment]
Please fix relnotes:

> Fox of a nil pointer dereference if `gracePeriod` is nil

Fix a nil pointer dereference in api server when `gracePeriod` was set to nil.

/lgtm

**Comment by disc** [comment]
> Please fix relnotes:
> 
> > Fox of a nil pointer dereference if `gracePeriod` is nil
> 
> Fix a nil pointer dereference in api server when `gracePeriod` was set to nil.
> 
> /lgtm

Fixed

**Comment by disc** [comment]
Looks like the issue has been fixed in another PR #127094, and my PR is no longer relevant.

**Pull request closed without merging**